commit/merkleroot/rmn/types/config.go

```diff
@@ -30,17 +30,29 @@ type RMNHomeNodeInfo struct {
 
 // RMNRemoteConfig contains the configuration fetched from the RMNRemote contract.
 type RMNRemoteConfig struct {
-	ContractAddress  cciptypes.Bytes
-	ConfigDigest     cciptypes.Bytes32
-	Signers          []RMNRemoteSignerInfo
-	MinSigners       uint64
-	ConfigVersion    uint32
-	RmnReportVersion string // e.g., "RMN_V1_6_ANY2EVM_REPORT"
+	ContractAddress  cciptypes.Bytes       `json:"contractAddress"`
+	ConfigDigest     cciptypes.Bytes32     `json:"configDigest"`
+	Signers          []RMNRemoteSignerInfo `json:"signers"`
+	MinSigners       uint64                `json:"minSigners"`
+	ConfigVersion    uint32                `json:"configVersion"`
+	RmnReportVersion string                `json:"rmnReportVersion"` // e.g., "RMN_V1_6_ANY2EVM_REPORT"
+}
+
+func (r RMNRemoteConfig) IsEmpty() bool {
+	return len(r.ContractAddress) == 0 ||
+		r.ConfigDigest.IsEmpty() ||
+		len(r.Signers) == 0 ||
+		r.MinSigners == 0 ||
+		r.ConfigVersion == 0 ||
+		r.RmnReportVersion == ""
 }
 
 // RMNRemoteSignerInfo contains information about a signer from the RMNRemote contract.
 type RMNRemoteSignerInfo struct {
-	OnchainPublicKey      cciptypes.Bytes // The signer's onchain address, used to verify report signature
-	NodeIndex             uint64          // The index of the node in the RMN config
-	SignObservationPrefix string          // The prefix of the observation to sign
+	// The signer's onchain address, used to verify report signature
+	OnchainPublicKey cciptypes.Bytes `json:"onchainPublicKey"`
```

Comment:
Should be renamed I think `OnchainPublicKey` -> `SignerAddress` or something. Let's use a name similar to what the contract uses.

Reply:
Contract uses `onchainPublicKey` which is why I wanted to rename to this https://github.com/smartcontractkit/ccip/blob/7d4f8e56e27faba2007edff7b6623ac7c29ec94a/contracts/src/v0.8/ccip/rmn/RMNRemote.sol#L43

Reply:
Isn't that wrong though?